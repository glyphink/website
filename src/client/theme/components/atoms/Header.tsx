/**
 * @file Components - Header
 * @module website/theme/components/atoms/Header
 */

import type { Props } from '@glyphink/website/theme'
import {
  forwardRef,
  type ForwardedRef,
  type ForwardRefExoticComponent,
  type JSX
} from 'react'
import type HeaderProps from './Header.props.mts'

/**
 * Introductory content.
 *
 * @see https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/header
 * @see {@linkcode ForwardRefExoticComponent}
 * @see {@linkcode HeaderProps}
 *
 * @category
 *  atoms
 *
 * @const {ForwardRefExoticComponent<HeaderProps>} Header
 */
const Header: ForwardRefExoticComponent<HeaderProps> = forwardRef<
  HTMLElement,
  HeaderProps
>(
  /**
   * @param {Props<HeaderProps>} props
   *  Component props
   * @param {ForwardedRef<HTMLElement>} ref
   *  Forwarded ref
   * @return {JSX.Element}
   *  JSX `<header>` element
   */
  (props: Props<HeaderProps>, ref: ForwardedRef<HTMLElement>): JSX.Element => {
    return <header {...props} ref={ref} />
  }
)

Header.displayName = 'Header'

export default Header
