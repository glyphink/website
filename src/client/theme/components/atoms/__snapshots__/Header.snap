// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:theme/components/atoms/Header > should render <header> 1`] = `<header />`;
